SingularityUI/app/components/taskSearch/QueryParam.cjsx

```diff
@@ -0,0 +1,14 @@
+QueryParam = React.createClass
+
+    getClassName: ->
+        name="list-group-item"
+        if @props.cantClear
+            name += " disabled"
+        return name
+
+    render: ->
+        return  <li className={@getClassName()}>
+                        <b>{@props.paramName}:</b> {@props.paramValue} 
+                        {<span className="remove-query-param glyphicon glyphicon-remove pull-right" onClick={@props.onClick} /> unless @props.cantClear}
```

Comment:
As a rule, anything clickable should should a `<button>` tag. A clickable `<span>` can't be selected via the keyboard, and doesn't indicate its clickability when you hover over it with the mouse. If SingularityUI doesn't already have an "icon button" style class, I would add one.